### ISSUE TITLE
[Config Support]: SHM size wont go over 268.5MB despite log error recommendation

### ISSUE THREAD

**Issue by domenicdistefano** [blakeblackshear/frigate#7739]
### Describe the problem you are having

Doing some extreme / experimental testing with my 12 camera setup pushing the limits of some cameras and my config overall.

I calculated the total amount of memory required for my test scenario to be ~2GB so I edited my config.yml to read` shm_size: "2048mb"`. When I restart Frigate however I I receive this error in the logs: `The current SHM size of 268.4MB is too small, recommend increasing it to at least 674.6999999999999MB.` I couldn't find any artificial upper limit or other limitation stated in the documentation and the error messages suggests increasing to 674mb.  If I attempt to enter a much smaller # of say 300MB Frigate again arbitrarily sets the value to 268.4MB and displays the same error message. 

### Version

(0.12.1-367d724

### Frigate config file

```yaml
#ui:
#  use_experimental: false
#  live_mode: mse

mqtt:
  enabled: false

#record:
#  expire_interval: 10

go2rtc:
  streams:
    back_porch:
      - rtsp://admin:password@10.10.10.71:554/media/video0
    back_porch_sub:
      - rtsp://admin:password@10.10.10.71:554/media/video0
##############################################################################
    pool_road:
      - rtsp://admin:password@10.10.10.70:554/media/video0
    pool_road_sub:
      - rtsp://admin:password@10.10.10.70:554/media/video0
##############################################################################  
    back_yard:
      - rtsp://admin:password@10.10.10.72:554/media/video0
    back_yard_sub:
      - rtsp://admin:password@10.10.10.72:554/media/video0
##############################################################################
    garden:
      - rtsp://admin:password@10.10.10.73:554/media/video0
    garden_sub:
      - rtsp://admin:password@10.10.10.73:554/media/video0
##############################################################################
    front_yard:
      - rtsp://admin:password@10.10.10.74:554/media/video0
    front_yard_sub:
      - rtsp://admin:password@10.10.10.74:554/media/video0
##############################################################################
    front_door:
      - rtsp://admin:password@10.10.10.75:554/media/video0
    front_door_sub:
      - rtsp://admin:password@10.10.10.75:554/media/video0
##############################################################################
    driveway_front:
      - rtsp://admin:password@10.10.10.76:554/media/video0
    driveway_front_sub:
      - rtsp://admin:password@10.10.10.76:554/media/video0
##############################################################################
    driveway_pool:
      - rtsp://admin:password@10.10.10.77:554/media/video0
    driveway_pool_sub:
      - rtsp://admin:password@10.10.10.77:554/media/video0
##############################################################################
    lpr-41:
      - rtsp://admin:password@10.10.10.63:554/streaming/channels/101
    lpr-41_sub:
      - rtsp://admin:password@10.10.10.63:554/streaming/channels/101
##############################################################################
    lpr-cheltenham:
      - rtsp://admin:password@10.10.10.64:554/streaming/channels/101
    lpr-cheltenham_sub:
      - rtsp://admin:password@10.10.10.64:554/streaming/channels/101
##############################################################################
    lpr-pool:
      - rtsp://admin:password@10.10.10.65:554/streaming/channels/101
    lpr-pool_sub:
      - rtsp://admin:password@10.10.10.65:554/streaming/channels/101
##############################################################################
    doorbell: 
      - "ffmpeg:http://10.10.10.60/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=admin&password=dingdong#video=copy#audio=copy#audio=opus"
    doorbell_sub: 
      - "ffmpeg:http://10.10.10.60/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=admin&password=dingdong"


timestamp_style:
  # Optional: Position of the timestamp (default: shown below)
  #           "tl" (top left), "tr" (top right), "bl" (bottom left), "br" (bottom right)
  position: "tl"
  # Optional: Format specifier conform to the Python package "datetime" (default: shown below)
  #           Additional Examples:
  #             german: "%d.%m.%Y %H:%M:%S"
  format: "%m/%d/%Y %H:%M:%S"
  # Optional: Color of font
  color:
    # All Required when color is specified (default: shown below)
    red: 255
    green: 255
    blue: 255
  # Optional: Line thickness of font (default: shown below)
  thickness: 2
  # Optional: Effect of lettering (default: shown below)
  #           None (No effect),
  #           "solid" (solid background in inverse color of font)
  #           "shadow" (shadow for font)
  effect: shadow

cameras:

 ##############################################################################
  back_porch:
    ffmpeg:
#    hwaccel_args: preset-vaapi
      inputs:
        - path: rtsp://127.0.0.1:8554/back_porch_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
          hwaccel_args: preset-vaapi
        - path: rtsp://127.0.0.1:8554/back_porch
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 2560
      height: 1440
      fps: 20
    objects:
      track:
        - person
        - dog

      filters:
        dog:
          min_score: 0.3
          threshold: 0.5
#      mask:
#        - 23,480,333,480,333,442,26,440
#        - 640,95,640,0,464,0
#        - 258,211,323,246,357,142,274,140
#        - 383,342,457,365,477,261,401,232
#        - 425,131,553,227,640,251,640,213,607,176,460,88
    snapshots:
      enabled: true
      timestamp: false
      bounding_box: true
      retain:
        default: 3
  #  motion:
  #    mask:
  #      - 337,480,341,437,0,433,0,480,40,480

 #   zones:
 #     deck_area:
 #      coordinates: 0,284,0,457,207,480,474,480,640,480,540,406,392,259,253,106,166,31,108,49,0,72
    record:
      enabled: True
      retain: 
        days: 0
      events:
        retain:
          default: 3
          mode: motion 
        pre_capture: 5
        post_capture: 15  
##############################################################################
  pool_road:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/pool_road_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/pool_road
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 3840
      height: 2160
      fps: 30
    objects:
      track:
        - person
        - dog

      filters:
        dog:
          min_score: 0.3
          threshold: 0.5
#      mask:
#        - 23,480,333,480,333,442,26,440
#        - 640,95,640,0,464,0
#        - 258,211,323,246,357,142,274,140
#        - 383,342,457,365,477,261,401,232
#        - 425,131,553,227,640,251,640,213,607,176,460,88
    snapshots:
      enabled: true
      timestamp: false
      bounding_box: true
      retain:
        default: 3
  #  motion:
  #    mask:
  #      - 337,480,341,437,0,433,0,480,40,480

 #   zones:
 #     deck_area:
 #      coordinates: 0,284,0,457,207,480,474,480,640,480,540,406,392,259,253,106,166,31,108,49,0,72
    record:
      enabled: True
      retain: 
        days: 0
      events:
        retain:
          default: 3
          mode: motion 
        pre_capture: 5
        post_capture: 15
##############################################################################
  back_yard:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/back_yard_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
          hwaccel_args: preset-vaapi           
        - path: rtsp://127.0.0.1:8554/back_yard
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 2560
      height: 1440
      fps: 20
    objects:
      track:
        - person
        - dog

      filters:
        dog:
          min_score: 0.3
          threshold: 0.5
#      mask:
#        - 23,480,333,480,333,442,26,440
#        - 640,95,640,0,464,0
#        - 258,211,323,246,357,142,274,140
#        - 383,342,457,365,477,261,401,232
#        - 425,131,553,227,640,251,640,213,607,176,460,88
    snapshots:
      enabled: true
      timestamp: false
      bounding_box: true
      retain:
        default: 3
  #  motion:
  #    mask:
  #      - 337,480,341,437,0,433,0,480,40,480

 #   zones:
 #     deck_area:
 #      coordinates: 0,284,0,457,207,480,474,480,640,480,540,406,392,259,253,106,166,31,108,49,0,72
    record:
      enabled: True
      retain: 
        days: 0
      events:
        retain:
          default: 3
          mode: motion 
        pre_capture: 5
        post_capture: 15  
##############################################################################
  garden:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/garden_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/garden
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 2560
      height: 1440
      fps: 20
    objects:
      track:
        - person
        - dog

      filters:
        dog:
          min_score: 0.3
          threshold: 0.5
#      mask:
#        - 23,480,333,480,333,442,26,440
#        - 640,95,640,0,464,0
#        - 258,211,323,246,357,142,274,140
#        - 383,342,457,365,477,261,401,232
#        - 425,131,553,227,640,251,640,213,607,176,460,88
    snapshots:
      enabled: true
      timestamp: false
      bounding_box: true
      retain:
        default: 3
  #  motion:
  #    mask:
  #      - 337,480,341,437,0,433,0,480,40,480

 #   zones:
 #     deck_area:
 #      coordinates: 0,284,0,457,207,480,474,480,640,480,540,406,392,259,253,106,166,31,108,49,0,72
    record:
      enabled: True
      retain: 
        days: 0
      events:
        retain:
          default: 3
          mode: motion 
        pre_capture: 5
        post_capture: 15  
##############################################################################
  front_yard:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/front_yard_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/front_yard
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 3840
      height: 2160
      fps: 30
    objects:
      track:
        - person
        - dog

      filters:
        dog:
          min_score: 0.3
          threshold: 0.5
#      mask:
#        - 23,480,333,480,333,442,26,440
#        - 640,95,640,0,464,0
#        - 258,211,323,246,357,142,274,140
#        - 383,342,457,365,477,261,401,232
#        - 425,131,553,227,640,251,640,213,607,176,460,88
    snapshots:
      enabled: true
      timestamp: false
      bounding_box: true
      retain:
        default: 3
  #  motion:
  #    mask:
  #      - 337,480,341,437,0,433,0,480,40,480

 #   zones:
 #     deck_area:
 #      coordinates: 0,284,0,457,207,480,474,480,640,480,540,406,392,259,253,106,166,31,108,49,0,72
    record:
      enabled: True
      retain: 
        days: 0
      events:
        retain:
          default: 3
          mode: motion 
        pre_capture: 5
        post_capture: 15  
##############################################################################
  front_door:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/front_door_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
          hwaccel_args: preset-vaapi            
        - path: rtsp://127.0.0.1:8554/front_door
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 2560
      height: 1440
      fps: 20
    objects:
      track:
        - person
        - dog

      filters:
        dog:
          min_score: 0.3
          threshold: 0.5
#      mask:
#        - 23,480,333,480,333,442,26,440
#        - 640,95,640,0,464,0
#        - 258,211,323,246,357,142,274,140
#        - 383,342,457,365,477,261,401,232
#        - 425,131,553,227,640,251,640,213,607,176,460,88
    snapshots:
      enabled: true
      timestamp: false
      bounding_box: true
      retain:
        default: 3
  #  motion:
  #    mask:
  #      - 337,480,341,437,0,433,0,480,40,480

 #   zones:
 #     deck_area:
 #      coordinates: 0,284,0,457,207,480,474,480,640,480,540,406,392,259,253,106,166,31,108,49,0,72
    record:
      enabled: True
      retain: 
        days: 0
      events:
        retain:
          default: 3
          mode: motion 
        pre_capture: 5
        post_capture: 15  
##############################################################################
  driveway_front:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/driveway_front_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/driveway_front
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 2560
      height: 1440
      fps: 20
    objects:
      track:
        - person
        - dog

      filters:
        dog:
          min_score: 0.3
          threshold: 0.5
#      mask:
#        - 23,480,333,480,333,442,26,440
#        - 640,95,640,0,464,0
#        - 258,211,323,246,357,142,274,140
#        - 383,342,457,365,477,261,401,232
#        - 425,131,553,227,640,251,640,213,607,176,460,88
    snapshots:
      enabled: true
      timestamp: false
      bounding_box: true
      retain:
        default: 3
  #  motion:
  #    mask:
  #      - 337,480,341,437,0,433,0,480,40,480

 #   zones:
 #     deck_area:
 #      coordinates: 0,284,0,457,207,480,474,480,640,480,540,406,392,259,253,106,166,31,108,49,0,72
    record:
      enabled: True
      retain: 
        days: 0
      events:
        retain:
          default: 3
          mode: motion 
        pre_capture: 5
        post_capture: 15  
##############################################################################
  driveway_pool:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/driveway_pool_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/driveway_pool
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 2560
      height: 1440
      fps: 20
    objects:
      track:
        - person
        - dog

      filters:
        dog:
          min_score: 0.3
          threshold: 0.5
#      mask:
#        - 23,480,333,480,333,442,26,440
#        - 640,95,640,0,464,0
#        - 258,211,323,246,357,142,274,140
#        - 383,342,457,365,477,261,401,232
#        - 425,131,553,227,640,251,640,213,607,176,460,88
    snapshots:
      enabled: true
      timestamp: false
      bounding_box: true
      retain:
        default: 3
  #  motion:
  #    mask:
  #      - 337,480,341,437,0,433,0,480,40,480

 #   zones:
 #     deck_area:
 #      coordinates: 0,284,0,457,207,480,474,480,640,480,540,406,392,259,253,106,166,31,108,49,0,72
    record:
      enabled: True
      retain: 
        days: 0
      events:
        retain:
          default: 3
          mode: motion 
        pre_capture: 5
        post_capture: 15  
##############################################################################
  lpr-41:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/lpr-41_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/lpr-41
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 1920
      height: 1080
      fps: 30
    objects:
      track:
        - person
        - dog

      filters:
        dog:
          min_score: 0.3
          threshold: 0.5
#      mask:
#        - 23,480,333,480,333,442,26,440
#        - 640,95,640,0,464,0
#        - 258,211,323,246,357,142,274,140
#        - 383,342,457,365,477,261,401,232
#        - 425,131,553,227,640,251,640,213,607,176,460,88
    snapshots:
      enabled: true
      timestamp: false
      bounding_box: true
      retain:
        default: 3
  #  motion:
  #    mask:
  #      - 337,480,341,437,0,433,0,480,40,480

 #   zones:
 #     deck_area:
 #      coordinates: 0,284,0,457,207,480,474,480,640,480,540,406,392,259,253,106,166,31,108,49,0,72
    record:
      enabled: True
      retain: 
        days: 0
      events:
        retain:
          default: 3
          mode: motion 
        pre_capture: 5
        post_capture: 15  
##############################################################################
  lpr-cheltenham:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/lpr-cheltenham_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/lpr-cheltenham
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 1920
      height: 1080
      fps: 30
    objects:
      track:
        - person
        - dog

      filters:
        dog:
          min_score: 0.3
          threshold: 0.5
#      mask:
#        - 23,480,333,480,333,442,26,440
#        - 640,95,640,0,464,0
#        - 258,211,323,246,357,142,274,140
#        - 383,342,457,365,477,261,401,232
#        - 425,131,553,227,640,251,640,213,607,176,460,88
    snapshots:
      enabled: true
      timestamp: false
      bounding_box: true
      retain:
        default: 3
  #  motion:
  #    mask:
  #      - 337,480,341,437,0,433,0,480,40,480

 #   zones:
 #     deck_area:
 #      coordinates: 0,284,0,457,207,480,474,480,640,480,540,406,392,259,253,106,166,31,108,49,0,72
    record:
      enabled: True
      retain: 
        days: 0
      events:
        retain:
          default: 3
          mode: motion 
        pre_capture: 5
        post_capture: 15  
##############################################################################
  lpr-pool:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/lpr-pool_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/lpr-pool
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 1920
      height: 1080
      fps: 30
    objects:
      track:
        - person
        - dog

      filters:
        dog:
          min_score: 0.3
          threshold: 0.5
#      mask:
#        - 23,480,333,480,333,442,26,440
#        - 640,95,640,0,464,0
#        - 258,211,323,246,357,142,274,140
#        - 383,342,457,365,477,261,401,232
#        - 425,131,553,227,640,251,640,213,607,176,460,88
    snapshots:
      enabled: true
      timestamp: false
      bounding_box: true
      retain:
        default: 3
  #  motion:
  #    mask:
  #      - 337,480,341,437,0,433,0,480,40,480

 #   zones:
 #     deck_area:
 #      coordinates: 0,284,0,457,207,480,474,480,640,480,540,406,392,259,253,106,166,31,108,49,0,72
    record:
      enabled: True
      retain: 
        days: 0
      events:
        retain:
          default: 3
          mode: motion 
        pre_capture: 5
        post_capture: 15  
##############################################################################
  doorbell:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/doorbell_sub?video=copy
          input_args: preset-rtsp-restream
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/doorbell?video=copy&audio=aac
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 2560
      height: 1920
      fps: 20
    objects:
      track:
        - person
        - dog

      filters:
        dog:
          min_score: 0.3
          threshold: 0.5
#      mask:
#        - 23,480,333,480,333,442,26,440
#        - 640,95,640,0,464,0
#        - 258,211,323,246,357,142,274,140
#        - 383,342,457,365,477,261,401,232
#        - 425,131,553,227,640,251,640,213,607,176,460,88
    snapshots:
      enabled: true
      timestamp: false
      bounding_box: true
      retain:
        default: 3
  #  motion:
  #    mask:
  #      - 337,480,341,437,0,433,0,480,40,480

 #   zones:
 #     deck_area:
 #      coordinates: 0,284,0,457,207,480,474,480,640,480,540,406,392,259,253,106,166,31,108,49,0,72
    record:
      enabled: True
      retain: 
        days: 0
      events:
        retain:
          default: 3
          mode: motion 
        pre_capture: 5
        post_capture: 15  
##############################################################################
 
detectors:
  coral1:
    type: edgetpu
    device: pci:0
  coral2:
    type: edgetpu
    device: pci:1

ffmpeg:
   hwaccel_args: preset-nvidia-h264
#  hwaccel_args: preset-vaapi
rtmp:
  enabled: false

birdseye:
  enabled: False
  restream: False
  width: 1280 
  height: 720
  quality: 1
  mode: objects

live:
  height: 720
  quality: 1
```

### Relevant log output

```shell
The current SHM size of 268.4MB is too small, recommend increasing it to at least 674.6999999999999MB
```

### Operating system

Debian

### Install method

Docker Compose

### Coral version

PCIe

**Comment by NickM-27** [comment]
> Frigate again arbitrarily sets the value to 268.4MB and displays the same error message.

Frigate does not control this, this is controlled by docker. 

I ran a test on my device and shm set to 2GB without issue (though I don't think that much will be needed, given frigates internal calculation of 674)

**Comment by domenicdistefano** [comment]
This was a clean build from scratch so it should behave just like your box did I would think as its just a simply change to the compose file.

Take a look at this [short 1minute video](https://youtu.be/hQvqezxloP4) of me simply changing  shm_size number from 256 to 768, saving & restarting, then the log showing `WARNING : The current SHM size of 268.4MB is too small, recommend increasing it to at least 674.6999999999999MB.`

I can't figure out what I am possibly doing wrong..... Notice in the error message its reporting the size as being set to this precellular value of 268.4MB.

**Comment by NickM-27** [comment]
Right, it seems for some reason docker is not able to allocate the full amount for shm. Make sure docker and docker compose are fully up to date, otherwise might need to create an issue in a docker repo

**Comment by domenicdistefano** [comment]
Docker version = 24.06 (build date of **9/4/23**) & Compose shows v2.21.0 which is dated **8/30/23** in the release notes. Can't get much more up to date than that.

To confirm I did a `docker inspect frigate | grep -i shm` which returned  `"ShmSize": 268435456,`

Yes it seems cut & dry here that the container is somehow misreading / mishandling whats indicated & being passed from the Compose file. I will head over there and open an issue and update this one when I get more info & hopefully resolved. 

Can you tell me what version of Docker & Compose you have on your box where changing the setting worked? When you set it to 2GB did you have 2GB or 2048MB in between the quotes?


![dockerver](https://github.com/blakeblackshear/frigate/assets/7042586/fe19ef5f-671e-4ac9-acb0-7376974402fc)

**Comment by NickM-27** [comment]
I run unraid so I am using docker cli, just adding the `--shm-size=xYZ`

For example with `2GB` used:

![Screen Shot 2023-09-09 at 07 49 49 AM](https://github.com/blakeblackshear/frigate/assets/14866235/cda62b55-dd19-48ea-886d-eef81b0578ef)

**Comment by domenicdistefano** [comment]
I solved the issue by accident but I don't understand why or how this worked. At work we would call this PFM (pure Fing magic).

I simply deleted the container & image at the command line and then instead of re-running `docker compose up` in the directory containing the compose file I pasted the contents of the compose file into Portioner which created it as a stack. 

The SOB started right up reporting the correct memory. Same exact compose file. No idea why creating the container through Portainer magically made it recognize the requested memory all of a sudden.
